### PR TITLE
lpips loss and training script fixes

### PR DESCRIPTION
**Pull request by pcuenca**

### CODE REVIEW

File: training/train_vit_vqvae.py [hunk 1]
```diff
@@ -376,7 +406,7 @@ def compute_loss(params):
     train_time = 0
     epochs = tqdm(range(num_epochs), desc=f"Epoch ... (1/{num_epochs})", position=0)
 
-    wandb.init(config=parser.parse_args())
+    wandb.init(config=flat_args(model_args, data_args, training_args))
```
Comment (borisdayma) [inline training/train_vit_vqvae.py:409]: Oh there was an issue with previous method?

File: training/train_vit_vqvae.py [hunk 2]
```diff
@@ -221,6 +222,36 @@ def create_learning_rate_fn(
     return warmup_fn
 
 
+def get_config(model_args, data_args, training_args):
+    if model_args.config_name is None:
+        return ViTVQConfig(
+            image_size=data_args.image_size,
+            commitment_cost=training_args.commitment_cost,
```
Comment (borisdayma) [inline training/train_vit_vqvae.py:229]: I realize we don't use commitment cost anymore.
Also there are so many args that we may just want to enforce providing a config file later

**Review by borisdayma — COMMENTED** [review]
This is awesome!
I left minor comments but otherwise I'll just merge as-is

File: training/train_vit_vqvae.py (hunk 1 above)
Comment (pcuenca) [reply training/train_vit_vqvae.py:409]: Only when supplying the arguments using a json file rather than command line arguments, see https://github.com/patil-suraj/vit-vqgan/blob/main/training/train_vit_vqvae.py#L203. In that case the parser stops the program because it thinks the required arguments have not been supplied (and even if it didn't, wandb would not record the right values).

File: training/train_vit_vqvae.py (hunk 2 above)
Comment (pcuenca) [reply training/train_vit_vqvae.py:229]: Should we remove the commitment cost then?

Comment (borisdayma) [reply training/train_vit_vqvae.py:229]: Yeah and maybe all config we do through the config file a bit like in dalle-mini?

**Comment by borisdayma** [comment]
Awesome! Thanks Pedro
